src/python/pants/backend/python/dependency_inference/rules.py

```diff
@@ -242,7 +278,73 @@ async def infer_python_dependencies_via_imports(
                 "One or more unowned dependencies detected. Check logs for more details."
             )
 
-    return InferredDependencies(sorted(merged_result))
+
+@rule(desc="Inferring Python dependencies by analyzing source")
+async def infer_python_dependencies_via_source(
+    request: InferPythonImportDependencies,
+    python_infer_subsystem: PythonInferSubsystem,
+    python_setup: PythonSetup,
+) -> InferredDependencies:
+    if (
+        not python_infer_subsystem.imports
+        and not python_infer_subsystem.resources_and_files_from_strings
+    ):
+        return InferredDependencies([])
+
+    _wrapped_tgt = await Get(WrappedTarget, Address, request.sources_field.address)
+    tgt = _wrapped_tgt.target
+    parsed_dependencies = await Get(
+        ParsedPythonDependencies,
+        ParsePythonDependenciesRequest(
+            cast(PythonSourceField, request.sources_field),
+            InterpreterConstraints.create_from_targets([tgt], python_setup),
+            string_imports=python_infer_subsystem.string_imports,
+            string_imports_min_dots=python_infer_subsystem.string_imports_min_dots,
+            string_resources=python_infer_subsystem.resources_and_files_from_strings,
+            string_resources_min_slashes=python_infer_subsystem.resources_and_files_from_strings_min_slashes,
+        ),
+    )
+
+    inferred_deps: set[Address] = set()
+    unowned_imports: set[str] = set()
+    parsed_imports = parsed_dependencies.imports
+    parsed_resources = parsed_dependencies.resources
+    if not python_infer_subsystem.imports:
+        parsed_imports = ParsedPythonImports([])
+
+    if parsed_imports:
```

Comment:
This is awkward. As far as settings goes we have `imports`, `string-imports` (which is ignored is `imports` is `False`) and this new one.

I'm wondering if we should pass `imports` into the `dependency_parser`, so it becomes the one-and-only funnel.